detox/src/devices/drivers/android/genycloud/GenyCloudDriver.js

```diff
@@ -17,43 +17,43 @@ class GenyCloudDriver extends AndroidDriver {
 
     const exec = new GenyCloudExec();
     const instanceNaming = new InstanceNaming(); // TODO should consider a permissive impl for debug/dev mode. Maybe even a custom arg in package.json (Detox > ... > genycloud > sharedAccount: false)
-    this.recipeService = new RecipesService(exec, logger);
-    this.instanceLookupService = new InstanceLookupService(exec, instanceNaming, this.deviceRegistry);
-    this.instanceLifecycleService = new InstanceLifecycleService(exec, instanceNaming);
-    this.deviceQueryHelper = new DeviceQueryHelper(this.recipeService);
-    this.allocationHelper = new AllocationHelper(this.instanceLookupService, this.instanceLifecycleService);
+    this.deviceRegistry = new GenyDeviceRegistryWrapper(this.deviceRegistry);
```

Comment:
The wrapper is installed as an agnostic adaptor, performing translations whenever the device-registry is used in common code.